src/components/user.js

```diff
@@ -15,14 +17,28 @@ export default class User extends React.Component {
         const { info = {}, address = {}, company = {} } = this.props;
 
         this.state = {
-            currentUser: {
+            currentUser:  {
                 info,
                 address,
                 company
             },
-            showAddress: false,
-            showCompany: false,
-            wantEdit:    false
+            formErrors:   {
+                name:      '',
+                email:     '',
+                phone:     '',
+                duplicate: '',
+                href:      ''
+            },
+            valid:        {
+                email: true,
```

Comment:
Can we try to get rid of `valid`?
We can use only `formErrors` object.
If some field has an error than the corresponding property will have a string (error message) otherwise it can be an empty line or `null`/`void 0`